### PR TITLE
Provide a functioning `getAuthenticatedRequest()` for Instagram

### PR DESCRIPTION
**Pull request by ramsey**
**TL;DR:** This adds the `access_token` query string parameter to the URL in order to return a properly authenticated Instagram request. More background details follow.

---

AbstractProvider provides a basic `getAuthenticatedRequest()` method, which the Instagram provider was using unchanged. This method creates a request, and in doing so, merges in the headers array returned by `getAuthorizationHeaders()`, which is an empty array in the AbstractProvider.

The base package provides a BearerAuthorizationTrait that providers may use in order to override `getAuthorizationHeaders()` and set an Authorization header using Bearer token authentication on all requests, in order to request a properly authenticated request.

Instagram, however, does not support the use of the Authorization header for authenticated requests. Instead, the `access_token` parameter must be present in the query string of all authenticated requests.

### CODE REVIEW

**Comment by ramsey** [comment]
@stevenmaguire Have you been able to take a look at this yet?

**Comment by stevenmaguire** [comment]
@ramsey Thanks for the bump! I'll take a look now.

**Comment by ramsey** [comment]
Thanks!
